src/metabase/warehouses/provider_detection.clj

```diff
@@ -0,0 +1,54 @@
+(ns metabase.warehouses.provider-detection
+  "Database provider detection based on hostname patterns."
+  (:require
+   [clojure.string :as str]))
+
+(def ^:private providers
+  "Database providers with their hostname patterns.
+   Based on frontend/src/metabase/databases/components/DatabaseHostnameWithProviderField/database-providers.ts"
+  [{:name "Aiven" :pattern #"\.aivencloud\.com$"}
+   {:name "Amazon RDS" :pattern #"\.rds\.amazonaws\.com$"}
+   {:name "Azure" :pattern #"\.postgres\.database\.azure\.com$"}
+   {:name "Crunchy Data" :pattern #"\.db\.postgresbridge\.com$"}
+   {:name "DigitalOcean" :pattern #"db\.ondigitalocean\.com$"}
+   {:name "Fly.io" :pattern #"\.fly\.dev$"}
+   {:name "Neon" :pattern #"\.neon\.tech$"}
+   {:name "PlanetScale" :pattern #"\.psdb\.cloud$"}
+   {:name "Railway" :pattern #"\.railway\.app$"}
+   {:name "Render" :pattern #"\.render\.com$"}
+   {:name "Scaleway" :pattern #"\.scw\.cloud$"}
+   {:name "Supabase" :pattern #"pooler\.supabase\.com|\.supabase\.co$"}
+   {:name "Timescale" :pattern #"\.tsdb\.cloud|\.timescale\.com$"}])
```

Comment:
Does the binding of `$` in the Timescale regex apply to both patterns joined by the `|` or just the latter one?